src/graphics/program-lib/chunks/lit/frag/lightmapDirAdd.js

```diff
@@ -11,7 +11,7 @@ void addLightMap() {
 
         dDiffuseLight += dLightmap * nlight * 2.0;
     }
-
-    dSpecularLight += dLightmap * getLightSpecular();
+    vec3 halfDir = normalize(normalize(-dLightDirNormW) + normalize(dViewDirW));
```

Comment:
I believe both of these are already normalised?

Reply:
Yes, that is right. I think the same mistake exists in clusteredLights.js, fixing it too.